client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriterBuffer.java

```diff
@@ -33,6 +33,7 @@ public class WriterBuffer {
   private List<WrappedBuffer> buffers = Lists.newArrayList();
   private int dataLength = 0;
   private int memoryUsed = 0;
+  private boolean isAvailable = false;
```

Comment:
Is it necessary？

Reply:
No. Buffer will only be access in a single thread.

Reply:
Why do we need this?

Reply:
When adding record, the memory is not sufficient. This will trigger the spill operation. Once buffer is flushed to remote server, the current buffer held by the writer should be dropped. So this is a mark for write to analyze whether it is available.

Reply:
Do we change this value in the spill thread?

Reply:
No, it won't

Any buffer flush operation in spill function only will do in the adding record thread.

Reply:
Remove this part.